specification/chaos/resource-manager/Microsoft.Chaos/preview/2022-10-01-preview/types/experiments.json

```diff
@@ -0,0 +1,811 @@
+          "description": "ISO8601 formatted string that represents a duration.",
+          "pattern": "^P(\\d+Y)?(\\d+M)?(\\d+D)?(T(\\d+H)?(\\d+M)?(\\d+(\\.\\d+)?S)?)?$"
+        },
+        "parameters": {
+          "description": "List of key value pairs.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/keyValuePair"
+          },
+          "x-ms-identifiers": [
+            "key"
+          ]
+        },
+        "selectorId": {
+          "type": "string",
+          "description": "String that represents a selector.",
+          "minLength": 1
+        }
+      },
+      "required": [
+        "duration",
+        "parameters",
+        "selectorId"
+      ],
+      "additionalProperties": false,
+      "x-ms-discriminator-value": "continuous"
+    },
+    "keyValuePair": {
+      "description": "A map to describe the settings of an action.",
+      "type": "object",
+      "properties": {
+        "key": {
+          "description": "The name of the setting for the action.",
+          "type": "string",
+          "minLength": 1
+        },
+        "value": {
+          "description": "The value of the setting for the action.",
+          "type": "string",
+          "minLength": 1
+        }
+      },
+      "required": [
+        "key",
+        "value"
+      ],
+      "additionalProperties": false
+    },
+    "selector": {
+      "type": "object",
+      "description": "Model that represents a selector in the Experiment resource.",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Enum of the selector type.",
+          "enum": [
+            "Percent",
+            "Random",
+            "Tag",
+            "List"
+          ],
+          "x-ms-enum": {
+            "name": "SelectorType",
+            "modelAsString": false
+          }
+        },
+        "id": {
+          "type": "string",
+          "description": "String of the selector ID.",
+          "minLength": 1
+        },
+        "targets": {
+          "type": "array",
+          "description": "List of Target references.",
+          "minItems": 1,
+          "items": {
+            "$ref": "#/definitions/targetReference"
+          }
+        },
+        "filter": {
+          "$ref": "#/definitions/filter"
+        }
+      },
+      "required": [
+        "type",
+        "id",
+        "targets"
+      ],
+      "additionalProperties": false
+    },
+    "filter": {
+      "type": "object",
+      "description": "Model that represents available filter types that can be applied to a targets list.",
+      "discriminator": "type",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Enum that discriminates between filter types. Currently only `Simple` type is supported.",
+          "enum": [
+            "Simple"
+          ],
+          "x-ms-enum": {
+            "name": "FilterType",
+            "modelAsString": false
```

Comment:
>modelAsString": false [](http://example.com/codeflow?start=13&length=21)

modelAsString: true is recommended. With a false value, it means that any change to this enum (including adding values) requires a new API version. If you use modelAsString: true, you can add new values without moving to a new API version. Ref: [https://armwiki.azurewebsites.net/rpaas/docs/fe/api-compatibility/schemaevolutionsupport-pre-adl.html?q=modelAsString#allowed-openapi-spec-updates-to-maintain-compatibility-and-avoid-data-corruption](https://armwiki.azurewebsites.net/rpaas/docs/fe/api-compatibility/schemaevolutionsupport-pre-adl.html?q=modelAsString#allowed-openapi-spec-updates-to-maintain-compatibility-and-avoid-data-corruption).

Reply:
Good call - updating.